_posts/2022-01-27-msbuild.md

```diff
@@ -0,0 +1,82 @@
+
+* `.gitignore` and `.gitattributes`
+* Standard license file
+* `nuget.config`: We use [our own NuGet feed](https://www.myget.org/F/particular/api/v3/index.json) to publish all sorts of pre-release packages and RTM packages before pushing them to nuget.org. This file ensures that all our projects can directly find and include packages pushed to our feed without the need to pollute nuget.org. If you want to get RTM releases a few minutes earlier or work with our pre-release packages, you can also use our feed.
+* virus-scan workflow: This is a [GitHub Action](https://github.com/features/actions) we use in all repositories to scan all releases with a virus scanner automatically.
+* dependabot configuration: Dependabot is (after its acquisition) a feature of GitHub that helps us update our (external and internal) dependencies across all repositories. With the number of repositories we have, Dependabot saves a tremendous effort by automating most dependency updates.
+* [Stale bot](https://github.com/apps/stale) configuration: This small GitHub app can close abandoned PRs and issues. It's easy to forget some older PRs with this many repositories after some time. Stale bot pings authors of PRs after a while of no progress on a PR and ultimately closes the PR if the issue doesn't seem to be urgent enough.
+* `.editorconfig`: We moved to [`.editorconfig`](https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options) to manage code style settings. All staff members have code style settings imported automatically and supported by major IDEs.
+* [Strong-naming](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming) certificates: The certificates are used to sign assemblies that we publish as NuGet packages (we sign the NuGet packages with a separate certificate, of course).
+* `Directory.Build.props`: This is what we're going to have a closer look at in this post.
+
+We use Azure Functions to sync these files on a schedule to every repository that contains a `.reposync.yml` file. An Azure Function using timed triggers checks all repositories that need to be updated. Because we have quite a few repositories to update, the actual work is handled separate. For every identified repository, the function uses NServiceBus to send a command via Azure ServiceBus (using [NServiceBus for Azure Functions](https://docs.particular.net/nservicebus/hosting/azure-functions-service-bus/)) to a message handler that will update a specific repository. For updates, we use the [GitHubSync](https://github.com/SimonCropp/GitHubSync) open source project from [@SimonCropp](https://twitter.com/SimonCropp) that takes care of the details. The `.reposync.yml` isn't just an empty marker file, it can also be used to exclude specific files from being synchronized, in case a repository needs to override the default or doesn't want this file. The content of a `.reposync` file might look like this (although most of them are empty):
```

Comment:
```suggestion
We use Azure Functions to sync these files on a schedule to every repository that contains a `.reposync.yml` file. An Azure Function using timed triggers checks all repositories that need to be updated. Because we have quite a few repositories to update, the synchronization work is handled separately. For every identified repository, the function uses NServiceBus to send a command via Azure ServiceBus (using [NServiceBus for Azure Functions](https://docs.particular.net/nservicebus/hosting/azure-functions-service-bus/)) to a message handler that will update a specific repository. For updates, we use the [GitHubSync](https://github.com/SimonCropp/GitHubSync) open-source project from [@SimonCropp](https://twitter.com/SimonCropp) that takes care of the details. The `.reposync.yml` isn't just an empty marker file. It can also be used to exclude specific files from being synchronized if a repository needs to override the default or doesn't want this file. The content of a `.reposync` file might look like this (although most of them are empty):
```